### PR TITLE
enable to do multiple commands/API calls using "##" on DELETE method

### PR DESCRIPTION
**Pull request by vmonnier**

### CODE REVIEW

File: adapters/cisco_nx_rest/cisco_nx_rest_command.php [hunk 1]
```diff
@@ -160,23 +160,36 @@ function eval_DELETE() {
...
+                        $xml_conf_str = trim ( $delete->evaluate_xml () );
+                        $xml_conf_str = str_replace ( "\n", '', $xml_conf_str );
+                        $xml_configs = explode ( "##", $xml_conf_str );
+
+                        if (! empty ( $operation_str )) {
+                                 if (count ( $xpaths ) != count ( $operations )) {
+                                        throw new SmsException ( "Operations are not as many as Xpaths" );
+                                } else
+                                                                {
+                                  $i = 0;
```
Comment (UBiqube-ydu) [inline adapters/cisco_nx_rest/cisco_nx_rest_command.php:178]: $i is not used

File: adapters/cisco_nx_rest/cisco_nx_rest_command.php [hunk 2]
```diff
@@ -160,23 +160,36 @@ function eval_DELETE() {
...
+                                  foreach ( $operations as $operation )
+                                  {
+                                        $conf = $operation . '##' . $xpaths[0];
```
Comment (UBiqube-ydu) [inline adapters/cisco_nx_rest/cisco_nx_rest_command.php:181]: $xpaths[$i] ?

File: adapters/cisco_nx_rest/cisco_nx_rest_command.php [hunk 3]
```diff
@@ -160,23 +160,36 @@ function eval_DELETE() {
...
+                                        $xml_conf_str = str_replace("\n", '', $xml_conf[0]);
```
Comment (UBiqube-ydu) [inline adapters/cisco_nx_rest/cisco_nx_rest_command.php:182]: $xml_configs[$i] ?

File: adapters/cisco_nx_rest/cisco_nx_rest_command.php [hunk 4]
```diff
@@ -160,23 +160,36 @@ function eval_DELETE() {
...
+                                        if (! empty ( $xml_conf_str )) {
+                                                 $conf .= "' -d'".$xml_conf_str;
+                                        }
+                                        $this->configuration .= "{$conf}\n";
+                                        $SMS_RETURN_BUF .= "{$conf}\n";
```
Comment (UBiqube-ydu) [inline adapters/cisco_nx_rest/cisco_nx_rest_command.php:187]: $SMS_RETURN_BUF can be set outside of the loop :
$SMS_RETURN_BUF = $this->configuration;

File: adapters/cisco_nx_rest/cisco_nx_rest_command.php (hunk 1 above)
Comment (yguubiqube) [reply adapters/cisco_nx_rest/cisco_nx_rest_command.php:178]: $i is used, and must be initialized to 0:
`$conf = $operation . '##' . $xpaths[$i];`
`$xml_conf_str = str_replace("\n", '', $xml_conf[$i]);`

File: adapters/cisco_nx_rest/cisco_nx_rest_command.php [hunk 5]
```diff
@@ -160,23 +160,37 @@ function eval_DELETE() {
...
+                                                 $conf .= "' -d'".$xml_conf_str;
+                                        }
+                                        $this->configuration .= "{$conf}\n";
+                                       
+                                        $i += 1;
+                                  }
+                                }
                         }
-		}
+                }
```
Comment (yguubiqube) [inline adapters/cisco_nx_rest/cisco_nx_rest_command.php:192]: I would have written:
```
                        if (! empty ( $operation_str )) {
                            if (count ( $xpaths ) != count ( $operations )) {
                                throw new SmsException ( "Operations are not as many as Xpaths" );
                            } else {
                                foreach ( $operations as $i => $operation )
                                {
                                    $conf = $operation . '##' . $xpaths[$i];
                                    $xml_conf_str = str_replace("\n", '', $xml_conf[$i]);
                                    if (! empty ( $xml_conf_str )) {
                                        $conf .= "' -d'".$xml_conf_str;
                                    }
                                    $this->configuration .= "{$conf}\n";
                                }
                            }   
                        }
                }
		$SMS_RETURN_BUF = $this->configuration;

```

**Review by yguubiqube — CHANGES_REQUESTED** [review]

File: adapters/cisco_nx_rest/cisco_nx_rest_command.php [hunk 6]
```diff
@@ -160,23 +160,37 @@ function eval_DELETE() {
...
+                        if (! empty ( $operation_str )) {
+                                 if (count ( $xpaths ) != count ( $operations )) {
+                                        throw new SmsException ( "Operations are not as many as Xpaths" );
+                                } else
+                                                                {
+                                  $i = 0;
+                                  foreach ( $operations as $operation )
+                                  {
+                                        $conf = $operation . '##' . $xpaths[$i];
+                                        $xml_conf_str = str_replace("\n", '', $xml_conf[$i]);
```
Comment (UBiqube-ydu) [inline adapters/cisco_nx_rest/cisco_nx_rest_command.php:182]: $xml_configs[$i] ?

**Review by UBiqube-ydu — APPROVED** [review]

**Pull request merged**